docs/topics/db/queries.txt

```diff
@@ -669,6 +669,36 @@ The ``F()`` objects support bitwise operations by ``.bitand()``, ``.bitor()``,
 
     Support for ``.bitxor()`` was added.
 
+.. _using-lookups-transforms-in-expressions:
+
+Expression can reference lookups and transforms
+-----------------------------------------------
+
+.. versionadded: 3.2
+
+Django also supports using lookups and transforms in expressions.
+
+This example finds all ``Entry`` objects published in the same year as they
+were last modified:
+
+    >>> Entry.objects.filter(pub_date__year=F('mod_date__year'))
+
+This example finds the earliest year an entry was published:
+
+    >>> Entry.objects.aggregate(first_published_year=Min('pub_date__year'))
+
+This example finds the value of the highest rated entry and the total number
+of comments on all entries for each year:
+
+    >>> Entry.objects.values('pub_date__year').annotate(
+            top_rating=Subquery(
+                Entry.objects.filter(
+                    pub_date__year=OuterRef('pub_date__year')
+                ).order_by('-rating').values('rating')[:1]
+            ),
+            total_comments=Sum('number_of_comments'),
+        )
```

Comment:
I don't entirely like this example - it's a bit more complicated than would be ideal.

I considered doing just:
```python
    >>> Entry.objects.annotate(
            top_rating=Subquery(
                Entry.objects.filter(
                    pub_date__year=OuterRef('pub_date__year')
                ).order_by('-rating').values('rating')[:1]
            ),
        )
```

But I don't like that this leads to duplicate rows in the resulting queryset - we get one for each `Entry` sharing `pub_date__year`. The example with `Sum` consolidates these into just one row.

I'm not sure explaining these nuances is in scope here - I'd prefer an example that completely avoids them.